src/Hashvatar.elm

```diff
@@ -0,0 +1,117 @@
+module Hashvatar exposing (..)
+
+import Hash exposing (Hash)
+import Hashvatar.HexGrid as HexGrid
+import Html exposing (Html, div)
+import Html.Attributes exposing (class)
+import List.Extra as ListE
+import String.Extra exposing (break)
+import UI.Color as Color exposing (Color)
+
+
+view : Hash -> Html msg
+view hash =
+    let
+        raw =
+            hash |> Hash.toString |> String.replace "#" ""
+
+        numSlots =
+            5
+
+        partLength =
+            String.length raw // numSlots
+
+        parts =
+            break partLength raw
+
+        toCharCodeSum str =
+            str
+                |> String.toList
+                |> List.foldl (\c acc -> acc + Char.toCode c) 0
+
+        grid =
+            parts
+                |> List.map toCharCodeSum
+                |> toGrid
+                |> Maybe.withDefault HexGrid.empty
+    in
+    div [ class "hashvatar" ]
+        [ HexGrid.view grid
+        ]
+
+
+
+-- Helpers
+
+
+getIn : Int -> List Color -> Maybe Color
+getIn unmoddedIdx colors_ =
+    ListE.getAt (modBy (List.length colors_) unmoddedIdx) colors_
+
+
+toGrid : List Int -> Maybe HexGrid.HexGrid
+toGrid slots =
```

Comment:
This function feels a bit gnarly. Its mostly due to having to deal with optionality when getting elements in the list by index, even though we know the size of the list with a guarantee.